### ISSUE TITLE
bcrypt is broken on canary version of deno

### ISSUE THREAD

**Issue by ralyodio** [JamesBroadberry/deno-bcrypt#27]
Use `deno upgrade --canary` and the `bcrypt.compare(pw, hashed_pw)` function no longer works on accounts created before upgrading. Accounts created using canary DO work.

If you downgrade back to 1.22 `deno upgrade` (remove --canary flag) older accounts can now login but newer accoutns created on --canary can NOT login anymore.

**Comment by JamesBroadberry** [comment]
I can't seem to replicate what's going on. I've installed Deno 1.22 and generated a hash then upgraded to Deno Canary (version details below) and compared that hash with the original plaintext and it's validated. Also, all tests continue to pass on Canary (which test a pre-existing hash)

Could you provide some more specific (simple) steps to replicate what you're seeing?

Plaintext: `TestPassword123`
Hash generated in Deno 1.22: `$2a$10$kdKOsHZY1/7MAuejHnMZneGbW14YTs4CYDQr9Zb3tMi0YJ92syYyG`
Valid in Deno Canary: `true`

deno 1.22.0+d55444b (canary, x86_64-pc-windows-msvc)
v8 10.0.139.17
typescript 4.6.2

**Comment by ralyodio** [comment]
so for me. if you generate hash on 1.22 (stable) and the try to compare on 1.22 (canary) it returns false.

**Comment by JamesBroadberry** [comment]
Closed as cannot replicate

**Comment by JamesBroadberry** [comment]
Hi @ralyodio, sorry I couldn't replicate this here. I've fixed the root cause under #28, check the discussion there for details about why it happened and how it's fixed. In the meantime, for any production apps, either use < Deno 1.23.0 with BCrypt v0.3.0 or lower or upgrade to BCrypt v0.4.0 (with any Deno version). 

tl;dr [Deno 1.23.0 introduced a bug in the TypeScript transpilation](https://github.com/denoland/deno/issues/14900) that essentially caused the cipher to be bypassed.
